File: haskell/src/Domain/Task.hs
module Domain.Task where

import Data.Text (Text)
import Domain.User (FamilyId, UserId)

newtype TaskId = TaskId Int deriving (Show, Eq)

data TaskStatus = Pending | Completed | Approved | Rejected deriving (Eq, Show)

data Task = Task
  { taskId :: TaskId,
    familyId :: FamilyId,
    assignedTo :: UserId,
    title :: Text,
    pointValue :: Int,
    status :: TaskStatus
  }
  deriving (Eq, Show)

File: ocaml/lib/task.ml
type task_id = TaskId of int
type task_status = Pending | Completed | Approved | Rejected

type t = {
  id : task_id;
  family_id : User.family_id;
  assigned_to : User.user_id;
  title : string;
  point_value : int;
  status : task_status;
}
